swift/ql/test/query-tests/Security/CWE-328/testCrypto.swift

```diff
@@ -21,48 +52,119 @@ enum Insecure {
     }
 }
 
-func test1(passwd : UnsafeRawBufferPointer, encrypted_passwd : String, account_no : String, credit_card_no : String) {
+func testHashMethods(passwd : UnsafeRawBufferPointer, cert: String, encrypted_passwd : String, account_no : String, credit_card_no : String) {
     var hash = Crypto.Insecure.MD5.hash(data: passwd)  // BAD
+    hash = Crypto.Insecure.MD5.hash(data: cert)   // BAD [NOT DETECTED]
     hash = Crypto.Insecure.MD5.hash(data: encrypted_passwd)  // GOOD  (not sensitive)
     hash = Crypto.Insecure.MD5.hash(data: account_no)   // BAD [NOT DETECTED]
     hash = Crypto.Insecure.MD5.hash(data: credit_card_no)   // BAD
-}
+    hash = Crypto.Insecure.MD5.hash(data: credit_card_no)   // BAD
 
-func test2(passwd : String, encrypted_passwd : String, account_no : String, credit_card_no : String) {
-    var hash = Crypto.Insecure.SHA1.hash(data: passwd)  // BAD
+    hash = Crypto.Insecure.SHA1.hash(data: passwd)  // BAD
+    hash = Crypto.Insecure.SHA1.hash(data: cert)   // BAD [NOT DETECTED]
     hash = Crypto.Insecure.SHA1.hash(data: encrypted_passwd)  // GOOD  (not sensitive)
     hash = Crypto.Insecure.SHA1.hash(data: account_no)   // BAD [NOT DETECTED]
     hash = Crypto.Insecure.SHA1.hash(data: credit_card_no)   // BAD
+
+    hash = Crypto.SHA256.hash(data: passwd)   // BAD [NOT DETECTED] not a computationally hard hash
+    hash = Crypto.SHA256.hash(data: cert)   // GOOD
+    hash = Crypto.SHA256.hash(data: account_no)   // GOOD
+    hash = Crypto.SHA256.hash(data: credit_card_no)   // GOOD
+    hash = Crypto.SHA256.hash(data: credit_card_no)   // GOOD
+
+    hash = Crypto.SHA256.hash(data: passwd)   // BAD [NOT DETECTED] not a computationally hard hash
```

Comment:
This line looks like it should use `SHA384`?

Reply:
whoops